app/src/telnet.cpp

```diff
@@ -0,0 +1,822 @@
+		if (OrigEsplogger == NULL)
+		{
+			printf ("telnetLogger: No original logger defined - Restarting...\n");
+			esp_restart();
+		}
+		esp_log_set_vprintf(OrigEsplogger);
+		return ESP_OK;
+	}
+
+	OrigEsplogger(fmt, list);
+
+	int res = vsprintf(telnetBuffer, fmt, list);
+	telnet_esp32_printf(telnetBuffer);
+
+	return res;
+}
+
+static void recvData(int sock, uint8_t *buffer, size_t _size)
+{
+	size_t size = _size - 2;	// Every line will end with \r\n
+	char *ptr;
+	int i;
+
+	if (size == 0)
+	{
+		if (telnetLoggerActive)
+		{
+			telnet_esp32_printf ("Log monitoring disabled\n");
+			revertTelnetLogger();
+			ESP_LOGI (tag, "Log monitoring via telnet disabled");
+		}
+		return;
+	}
+
+	if (!wifiConnected())
+	{
+		ESP_LOGW(tag, "Wifi down! Aborting receive request!");
+		OperatingParameters.Errors.wifiErrors++;
+		return;
+	}
+	
+	buffer[size] = '\0';
+
+	ESP_LOGD (tag, "We received: %.*s", size, buffer);
+	switch(lookupCommnd((char *)buffer, size))
+	{
+		case HELP:
+			telnet_esp32_printf("Valid commands:\n");
+			telnet_esp32_printf("  Config:      Change configuration\n");
+			telnet_esp32_printf("  Help:        This!\n");
+			telnet_esp32_printf("  Up:          Increase set temp\n");
+			telnet_esp32_printf("  Down:        Decrease set temp\n");
+			telnet_esp32_printf("  Temp <temp>: Set arbitrary temp\n");
+			telnet_esp32_printf("  Mode <mode>: Set operating mode\n");
+			telnet_esp32_printf("  Monitor:     Monitor log output\n");
+			telnet_esp32_printf("  Status:      Dump status counters\n");
+			telnet_esp32_printf("  Error:       Dump error counters\n");
+			telnet_esp32_printf("  Quit:        Close telnet session\n");
+			telnet_esp32_printf("  Reboot:      Reboot the ESP32\n");
+			break;
+		case CONFIG:
+			doConfiguration(sock);
+			break;
+		case TEMP_UP:
+			telnet_esp32_printf ("Temperature up\n");
+			telnet_esp32_printf ("New set temp: %.1f\n", doTempUp());
+			break;
+		case TEMP_DOWN:
+			telnet_esp32_printf ("Temperature down\n");
+			telnet_esp32_printf ("New set temp: %.1f\n", doTempDown());
+			break;
+		case TEMP_SET:
+			ptr = strchr((char *)buffer, (int)' ');
+			ptr++;	// Move past ' '
+			i = atoi(ptr);
+			telnet_esp32_printf ("Set temperature to %d\n", i);
+			updateHvacSetTemp((float)i);
+			break;
+		case MODE_SET:
+		{
+			HVAC_MODE mode = ERROR;
+			telnet_esp32_printf ("Set mode\n");
+			ptr = strchr((char *)buffer, (int)' ');
+			ptr++;	// Move past ' '
+			ptr[0] = toupper(ptr[0]);
+			ESP_LOGI(tag, "Looking up %s", ptr);
+			mode = strToHvacMode(ptr);
+			if (mode == ERROR)
+			{
+				telnet_esp32_printf ("Invalid mode: %s\n", ptr);
+			}
+			else
+			{
+				updateHvacMode (mode);
+				telnet_esp32_printf ("Mode now set to: %s\n", ptr);
+			}
+		}
+			break;
+		case MONITOR_LOG:
+			telnet_esp32_printf ("Enable log monitoring (CR to exit)\n");
+			ESP_LOGI (tag, "Log monitoring via telnet enabled");
+			telnetLoggerActive = true;
+			OrigEsplogger = esp_log_set_vprintf(telnetLogger);
```

Comment:
issuing multiple `monitor` commands in a row over telnet will reliably crash the thermostat at the moment (3 `monitor`s in rapid succession crashes 100% of the time for me). Doing something like wrapping these lines up in
if (!telnetLoggerActive)
{
...
}
seems to fix it for me

Reply:
Yep! I can certainly see why!! I meant to address this before creating the PR and just forgot. So thanks for catching this.

I'm going to update the PR soon.

Reply:
Updated with eb02475235f6718518db407c99d5ed035d5fac44